jitterentropy-rngd: Update override syntax and SPDX license name

HatsyRei opened the pull request:
WI: [US#2829981](https://ni.visualstudio.com/DevCentral/_workitems/edit/2829981)

Building jitterentropy-rngd results in QA issues:
```
ERROR: jitterentropy-rngd-1.0+gitAUTOINC+23be1542a2-r0 do_package: QA Issue: File '/usr/sbin/jitterentropy-rngd' from jitterentropy-rngd was already stripped, this will prevent future debugging! [already-stripped]
ERROR: jitterentropy-rngd-1.0+gitAUTOINC+23be1542a2-r0 do_package: Fatal QA errors were found, failing task.
```

Turns out the old override syntax for `INSANE_SKIP` fails quietly, so the check still proceeds. 

Also convert to use SPDX license names in LICENSE variables, otherwise the build complains about it:
```
WARNING: jitterentropy-rngd-1.0+gitAUTOINC+23be1542a2-r0 do_package: QA Issue: jitterentropy-rngd: No generic license file exists for: BSD in any provider [license-exists]
WARNING: jitterentropy-rngd-1.0+gitAUTOINC+23be1542a2-r0 do_package_qa: QA Issue: Recipe LICENSE includes obsolete licenses GPLv2+ [obsolete-license]
```

chaitu236 reviewed (changes requested):
Can you please sign your commit message (git commit -s. Preferably using your ni/emerson email).

Also, please mention relevant Work Item # like in https://github.com/ni/openembedded-core/pull/132 if there is one.

HatsyRei commented:
@chaitu236 My apologies. Took a while to figure out the proper way to get a good signature.

gratian commented:
@HatsyRei can you clean up your branch and rebase/squash the commits you currently have in this PR? By policy, we are **not doing squash merges** on these open source repos so we want clean commit history.

This matches established upstream practice and keeps this repos maintainable. Remember any change submitted here will have to be maintained going forward for every future upgrade/rebase across all maintain branches/versions/products. Somebody in the future would otherwise have to look at 3 commits and try to understand what you were doing next time we rebase to a newer upstream version.

This brings me to the next point: please **include the why you made the change in the git commit message** itself not just the GitHub PR description (the "why'" is important, no need to explain "what" changed, that's usually obvious from the diff). A maintainer doing an upgrade/rebase in the future needs to be able to understand why the commit was added (and if it is still needed etc.) w/o referring to GitHub PR descriptions. It is not a given this repo will always be hosted on GitHub and tracking commits back to their originating PR is a time consuming and error prone process.

By the way the "Signed-off-by:" tag is still missing from your commit. Having this (developer certificate of origin) tag is required practice in open-source. When you do that you certify you have the right to submit this change to this open and public repo: https://developercertificate.org/. Like @chaitu236 already mentioned this will get automatically added for you if you use the '-s' option with 'git commit'.

HatsyRei committed:
jitterentropy-rngd: Fix override syntax, license name

jitterentropy-rngd is included in ARM target builds, which are currently
built via Yocto Sumo. Migrating these targets to Scarthgap will cause
builds to fail due to disruptive changes in override syntax introduced
in Honister. This commit fixes override syntax to allow it to build
correctly.

Recipe license specifications also replaced with SPDX identifiers to
avoid build warnings.

Signed-off-by: Hatsy Rei <yan.huan.chng@emerson.com>

HatsyRei commented:
@gratian Got it. Please let me know if there's anything wrong with the changes I've made so far.

chaitu236 commented:
> @gratian Got it. Please let me know if there's anything wrong with the changes I've made so far.

One last thing - we try to follow the [50/72 rule](https://stackoverflow.com/questions/2290016/git-commit-messages-50-72-formatting) i.e., subject line (i.e., first line) of commit message limited to 50 chars (if possible), empty line after that and rest of the commit message word-wrapped at 72 characters (except when there is quoted material or links, etc). You can refer to other commits in this repo for examples.
